File: app/src/main/java/com/cookplan/recipe/view_item/RecipeViewPresenter.java
package com.cookplan.recipe.view_item;

import com.cookplan.models.Ingredient;
import com.cookplan.models.ShopListStatus;

import java.util.List;

/**
 * Created by DariaEfimova on 23.03.17.
 */

public interface RecipeViewPresenter {

    public void getIngredientList();

    public void addIngredientToShoppingList(Ingredient ingredient);

    public void changeIngredListShopStatus(List<Ingredient> ingredients, ShopListStatus status);

    public void onStop();
}

File: app/src/main/java/com/cookplan/recipe/view_item/RecipeViewPresenter.kt
package com.cookplan.recipe.view_item

import com.cookplan.models.Ingredient
import com.cookplan.models.Recipe
import com.cookplan.models.ShopListStatus

/**
 * Created by DariaEfimova on 23.03.17.
 */

interface RecipeViewPresenter {

    fun getRecipeObject(): Recipe

    fun getIngredientList()

    fun addIngredientToShoppingList(ingredient: Ingredient)

    fun changeIngredListShopStatus(ingredients: List<Ingredient>, status: ShopListStatus)

    fun onStop()
}
